integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -75,97 +70,189 @@ def __init__(self, user, enterprise_configuration):
+                regardless of the last updated at time
+        """
+        ContentMetadataItemTransmission = apps.get_model(
+            'integrated_channel',
+            'ContentMetadataItemTransmission'
+        )
+        items_to_update = {}
+        for matched_item in matched_items:
+            content_query = Q(
+                enterprise_customer=self.enterprise_configuration.enterprise_customer,
+                integrated_channel_code=self.enterprise_configuration.channel_code(),
+                enterprise_customer_catalog_uuid=enterprise_customer_catalog.uuid,
+                content_id=matched_item.get('content_key'),
+                deleted_at__isnull=True,
+            )
+
+            # If not force_retrieve_all_catalogs, filter for content records where `content last changed` is less than
+            # the matched item's `date_updated`, otherwise select the row regardless of what the updated at time is.
+            last_changed_query = Q(content_last_changed__lt=matched_item.get('date_updated'))
+            last_changed_query.add(Q(content_last_changed__isnull=True), Q.OR)
+            if not force_retrieve_all_catalogs:
+                content_query.add(last_changed_query, Q.AND)
+
+            items_to_update_query = ContentMetadataItemTransmission.objects.filter(content_query)
+            item = items_to_update_query.first()
+            if item:
+                items_to_update[matched_item.get('content_key')] = item
+        return items_to_update
+
+    def _get_catalog_diff(self, enterprise_catalog, content_keys, force_retrieve_all_catalogs):
+        """
+        From the enterprise catalog API, request a catalog diff based off of a list of content keys. Using the diff,
+        retrieve past content metadata transmission records for update and delete payloads.
+        """
+        items_to_create, items_to_delete, matched_items = self.enterprise_catalog_api.get_catalog_diff(
+            enterprise_catalog,
+            content_keys
+        )
+        content_to_update = self._check_matched_content_updated_at(
+            enterprise_catalog,
+            matched_items,
+            force_retrieve_all_catalogs
+        )
+        content_to_delete = self._retrieve_past_transmission_content(
+            enterprise_catalog,
+            items_to_delete
+        )
+        return items_to_create, content_to_update, content_to_delete
+
+    def _retrieve_past_transmission_content(self, enterprise_customer_catalog, items):
+        """
+        Retrieve all past content metadata transmission records that have a `content_id` contained within a provided
+        list.
+        """
+        ContentMetadataItemTransmission = apps.get_model(
+            'integrated_channel',
+            'ContentMetadataItemTransmission'
+        )
+        content_keys = [item.get('content_key') for item in items]
+
+        # Do we want to filter out deleted items here?
```

Comment:
Is this still an open question?

Reply:
It was when writing, I think I am more confident in the state now.